src/main/java/org/spongepowered/common/mixin/core/entity/EntityMixin.java

```diff
@@ -558,21 +544,22 @@ private void onSpongeReadFromNBT(final CompoundNBT compound, final CallbackInfo
 
     }
 
-    @Redirect(method = "doBlockCollisions", at = @At(value = "INVOKE", target = "Lnet/minecraft/block/Block;onEntityCollision(Lnet/minecraft/world/World;Lnet/minecraft/util/math/BlockPos;Lnet/minecraft/block/state/IBlockState;Lnet/minecraft/entity/Entity;)V")) // doBlockCollisions
-    private void spongeImpl$onEntityCollideWithBlockState(
-        final Block block, final net.minecraft.world.World world, final BlockPos pos, final BlockState state, final Entity entity) {
+    @Redirect(method = "doBlockCollisions", at = @At(value = "INVOKE",
+        target = "Lnet/minecraft/block/BlockState;onEntityCollision(Lnet/minecraft/world/World;Lnet/minecraft/util/math/BlockPos;Lnet/minecraft/entity/Entity;)V")) // doBlockCollisions
+    private void spongeImpl$onEntityCollideWithBlockState(BlockState blockState, net.minecraft.world.World worldIn, BlockPos pos, Entity entityIn) {
         // if block can't collide, return
-        if (!((BlockBridge) block).bridge$hasCollideWithStateLogic()) {
+        if (!((BlockBridge) blockState.getBlock()).bridge$hasCollideWithStateLogic()) {
             return;
         }
 
         if (world.isRemote) {
-            block.onEntityCollision(world, pos, state, entity);
+            // In 1.14, why are we calling the state instead? You will notice these methods are deprecated on block since you are supposed to call the state method instead, the methods aren't actually deprecated.
```

Comment:
Leave out the comment. It's a recognized 1.14 change to call the methods on `BlockState` instead of the parent `Block`.